agent_api_rest/src/issuance/error.rs

```diff
@@ -0,0 +1,190 @@
+                .source(self)
+                .finish(),
+
+            // Public API Errors
+
+            // `/openid4vci/credential` endpoint
+            MissingCredentialDataError => ApiError::new(StatusCode::INTERNAL_SERVER_ERROR),
+        }
+    }
+}
+
+impl IntoApiErrorExt for OfferError {
+    fn into_api_error(self) -> ApiError {
+        use OfferError::*;
+
+        match self {
+            // UniCore API Problem Details
+            MissingCredentialOfferError => ApiError::builder(StatusCode::BAD_REQUEST)
+                .type_url(format!(
+                    "{DOCUMENTATION_URL}problem-details/issuance#missing-credential-offer"
+                ))
+                .title("Missing Credential Offer")
+                .source(self)
+                .finish(),
+            SendCredentialOfferError(_) => ApiError::builder(StatusCode::INTERNAL_SERVER_ERROR)
+                .type_url(format!(
+                    "{DOCUMENTATION_URL}problem-details/issuance#send-credential-offer-error"
+                ))
+                .title("Send Credential Offer Error")
+                .source(self)
+                .finish(),
+            InvalidCredentialOfferUriError(_) => ApiError::builder(StatusCode::INTERNAL_SERVER_ERROR)
+                .type_url(format!(
+                    "{DOCUMENTATION_URL}problem-details/issuance#invalid-credential-offer-uri"
+                ))
+                .title("Invalid Credential Offer URI")
+                .source(self)
+                .finish(),
+
+            // Public API Errors
+
+            // `/auth/token` endpoint
+            UnsupportedTokenRequestGrantTypeError => ApiError::new(StatusCode::INTERNAL_SERVER_ERROR),
+
+            // `/openid4vci/credential` endpoint
+            MissingCredentialError => ApiError::new(StatusCode::INTERNAL_SERVER_ERROR),
+            MissingProofError => ApiError::new(StatusCode::INTERNAL_SERVER_ERROR),
+            InvalidProofError(_) => ApiError::new(StatusCode::INTERNAL_SERVER_ERROR),
+            MissingProofIssuerError => ApiError::new(StatusCode::INTERNAL_SERVER_ERROR),
+        }
+    }
+}
+
+impl IntoApiErrorExt for ServerConfigError {
+    fn into_api_error(self) -> ApiError {
+        match self {}
+    }
+}
+
+#[cfg(test)]
+pub mod tests {
+    use super::*;
+    use crate::error::tests::into_json_value;
+    use serde_json::json;
+
+    #[tokio::test]
+    async fn issuance_errors_successfully_convert_to_problem_details() {
+        assert_eq!(
+            into_json_value(
+                CredentialError::UnsupportedCredentialFormat
+                    .into_api_error()
+                    .into_axum_response()
+            )
+            .await,
+            json!({
+                "type": format!("{DOCUMENTATION_URL}problem-details/issuance#unsupported-credential-format"),
+                "title": "Unsupported Credential Format",
+                "status": 500,
+                "detail": "This Credential format it not supported"
```

Comment:
Would it be easily possible to also include the value that the user provided? This way, a developer would be able to "close the loop" and search for that field in the request and/or config.

```
{
  "detail": "Credential format not supported: `dc+sd-jwt`"
}
```

Reply:
I've solved this, not in the prettiest way since I had to serialize the format via `json!` because `CredentialFormats`
doesn't implement the `Display` trait.

It will do for now I think until we remove this error variant altogether (since it should be a configuration error)

Reply:
I've solved this now, not in the prettiest way since I had to serialize the format via `json!` because `CredentialFormats`
doesn't implement the `Display` trait.

It will do for now I think until we remove this error variant altogether (since it should be a configuration error)